.github/workflows/publish.yml

```diff
@@ -38,6 +38,14 @@ jobs:
           CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
         shell: bash
         run: |
+          # Check if we can skip releasing a new version
+          # (there are no changes and the job was not manually triggered)
+          export CHANGED=$(cargo workspaces changed --ignore-changes "**/Cargo.toml")
```

Comment:
You might need `--include-merged-tags` here.

Reply:
I'll include it, but I don't think it's necessary since the publish job always executes on the master branch (this flag adds into consideration tags from merged branches).